testing/common.test.js

```diff
@@ -33,51 +37,54 @@ const execTestCafeCode = (t, code) => {
+  const getDemoPaths = (platform) => glob.sync('JSDemos/Demos/*/*')
+    .map((path) => join(path, platform));
 
-['jQuery', 'React', 'Vue', 'Angular'].forEach((approach) => {
-  const demoPaths = getDemoPaths(approach);
-  if (!shouldRunFramework(approach)) return;
+  getDemoPaths(approach).forEach((demoPath, index) => {
+    if (!shouldRunTestAtIndex(index) || !existsSync(demoPath)) { return; }
 
-  demoPaths.forEach((demoPath, index) => {
-    if (!shouldRunTestAtIndex(index)) return;
-    if (!existsSync(demoPath)) return;
+    function readFrom(relativePath, mapCallback) {
+      const absolute = join(demoPath, relativePath);
+      if (existsSync(absolute)) {
+        const result = readFileSync(absolute, 'utf8');
+        return (mapCallback && result && mapCallback(result)) || result;
+      }
+      return null;
+    }
 
     const testParts = demoPath.split('/');
     const widgetName = testParts[2];
     const demoName = testParts[3];
     const testName = `${widgetName}-${demoName}`;
 
-    const clientScriptPath = join(demoPath, '../client-script.js');
-    const testCodePath = join(demoPath, '../test-code.js');
-    const testCafeTestCodePath = join(demoPath, '../testcafe-test-code.js');
-    const visualTestSettingsPath = join(demoPath, '../visualtestrc.json');
+    const clientScriptSource = readFrom('../client-script.js', (x) => [{ content: x }]) || [];
+    const testCodeSource = readFrom('../test-code.js');
+    const testCafeCodeSource = readFrom('../testcafe-test-code.js');
+    const visualTestSettings = readFrom('../visualtestrc.json', (x) => JSON.parse(x));
 
-    const clientScriptSource = existsSync(clientScriptPath) ? [{ content: readFileSync(clientScriptPath, 'utf8') }] : [];
-    const testCodeSource = existsSync(testCodePath) ? readFileSync(testCodePath, 'utf8') : null;
-    const testCafeCodeSource = existsSync(testCafeTestCodePath) ? readFileSync(testCafeTestCodePath, 'utf8') : null;
-    const visualTestSettings = existsSync(visualTestSettingsPath) ? JSON.parse(readFileSync(visualTestSettingsPath, 'utf8')) : null;
+    if (process.env.GITHUB_CI) {
+      const approachLowerCase = approach.toLowerCase();
+      const ignoreApproach = visualTestSettings
+        && visualTestSettings[approachLowerCase]
+        && visualTestSettings[approachLowerCase].ignore;
 
-    const approachLowerCase = approach.toLowerCase();
-    const ignoreApproach = visualTestSettings
-      && visualTestSettings[approachLowerCase]
-      && visualTestSettings[approachLowerCase].ignore;
+      if (ignoreApproach) { return; }
+    }
 
-    if (ignoreApproach) return;
-    if (singleTestName && (testName !== singleTestName)) return;
-    (singleTestName ? test.only : test)
-      .page`http://127.0.0.1:808${getPortByIndex(index)}/JSDemos/Demos/${widgetName}/${demoName}/${approach}/`
+    runTestAt(test, `http://127.0.0.1:808${getPortByIndex(index)}/JSDemos/Demos/${widgetName}/${demoName}/${approach}/`)
```

Comment:
The runTestAt name does not say about specific for running. It will be better to use name runTestForDemo.

Reply:
`runTestAtPage`